File: day01/day01.py
from functools import reduce

# with open('input.txt', 'r') as myfile:
#   nums = [int(s) for s in myfile.read().rstrip()]
#   print(reduce(lambda sum, x: sum + x[0] if x[0] == x[1] else sum, zip(nums, nums[1:] + [nums[0]]), 0))
#   print(2 * reduce(lambda sum, x: sum + x[0] if x[0] == x[1] else sum, zip(nums[int(len(nums)/2):], nums[:int(len(nums)/2)]), 0))

with open('input.txt', 'r') as myfile:
  nums = [int(s) for s in myfile.read().rstrip()]

  total = 0
  for i in range(0, len(nums)):
    if nums[i] == nums[i - 1]:
      total += nums[i]

  print(total)

File: day01/day01.rb
input = File.read("input.txt").chomp.chars.map(&:to_i)
input2 = [input.last] + input[0..-2]
input3 = input[input.size/2..-1] + input[0...input.size/2]

puts input.zip(input2).reduce(0) { |acc, a| a[0] == a[1] ? acc + a[0] : acc }
puts input.zip(input3).reduce(0) { |acc, a| a[0] == a[1] ? acc + a[0] : acc }
